config/dependency.go

```
@@ -382,28 +386,100 @@ func cloneTerragruntOptionsForDependencyOutput(terragruntOptions *options.Terrag
+		terragruntOptions.Logger.Printf("WARNING: Could not parse remote_state block from target config %s", targetConfig)
+		terragruntOptions.Logger.Printf("WARNING: Falling back to terragrunt output.")
+		return runTerragruntOutputJson(targetTGOptions, targetConfig)
+	}
+	return getTerragruntOutputJsonFromRemoteState(targetTGOptions, targetConfig, remoteStateTGConfig.RemoteState)
+}
+
+// canGetRemoteState returns true if the remote state block is not nil and has the generate block configured.
+func canGetRemoteState(remoteState *remote.RemoteState) bool {
+	return remoteState != nil && remoteState.Generate != nil
+}
+
+// getTerragruntOutputJsonFromRemoteState will retrieve the outputs directly by using just the remote state block. This
+// uses terraform's feature where `output` and `init` can work without the real source, as long as you have the
+// `backend` configured.
+// To do this, this function will:
+// - Create a temporary folder
+// - Generate the backend.tf file with the backend configuration from the remote_state block
+// - Run terraform init and terraform output
+// - Clean up folder once json file is generated
+// NOTE: targetTGOptions should be in the context of the targetConfig.
+func getTerragruntOutputJsonFromRemoteState(targetTGOptions *options.TerragruntOptions, targetConfig string, remoteState *remote.RemoteState) ([]byte, error) {
+	util.Debugf(targetTGOptions.Logger, "Detected remote state block with generate config. Resolving dependency by pulling remote state.")
+
+	// Create working directory where we will run terraform in. Make sure it is cleaned up before the function returns.
+	tempWorkDir, err := ioutil.TempDir("", "")
+	if err != nil {
+		return nil, err
+	}
+	defer os.RemoveAll(tempWorkDir)
+
+	// Set the terraform working dir to the tempdir, and set stdout writer to ioutil.Discard so that output content is
+	// not logged.
+	targetTGOptions.WorkingDir = tempWorkDir
+	targetTGOptions.Writer = ioutil.Discard
```

Will this result in errors being hidden?

---

Nope because the `ErrWriter` is shared and all errors are writen to `stderr` so you will see the error messages if this command fails.